### ISSUE TITLE
Use istioctl install istio, when enabled components egressGateway and Values.global.sds.enabled was error

### ISSUE THREAD

**Issue by zhengqisong** [istio/istio#20199]
**Bug description**：Use istioctl install istio, when enabled components egressGateway and Values.global.sds.enabled was error.

**Affected product area (please put an X in all that apply)**：  Installation
**How was Istio installed?**: istioctl
**Environment where bug was observed (cloud vendor, OS, etc)**：kubernetes 1.15.4

**./bin/istioctl manifest generate  -f test output** 
```
volumes:
      - hostPath:
          path: /var/run/sds
        name: sdsudspath
      - name: istio-token
        projected:
          sources:
          - serviceAccountToken:
              audience: istio-ca
              expirationSeconds: 43200
              path: istio-token
      - hostPath:
          path: /var/run/sds
        name: sdsudspath
      - name: istio-certs
        secret:
          optional: true
          secretName: istio.default
      - name: egressgateway-certs
        secret:
          optional: true
          secretName: istio-egressgateway-certs
```
install/kubernetes/operator/charts/gateways/istio-egress/templates/deployment.yaml
```
{{ $gateway := index .Values "gateways" "istio-egressgateway" }}
{{- $labels := (dict "app" "istio-egressgateway" "istio" "egressgateway") }}
apiVersion: apps/v1
kind: Deployment
metadata:
  name: istio-egressgateway
  namespace: {{ .Release.Namespace }}
  labels:
{{ $labels | toYaml | indent 4 }}
    release: {{ .Release.Name }}
spec:
{{- if not $gateway.autoscaleEnabled }}
{{- if $gateway.replicaCount }}
  replicas: {{ $gateway.replicaCount }}
{{- end }}
{{- end }}
  selector:
    matchLabels:
      app: istio-egressgateway
      istio: egressgateway
  strategy:
    rollingUpdate:
      maxSurge: {{ $gateway.rollingMaxSurge }}
      maxUnavailable: {{ $gateway.rollingMaxUnavailable }}
  template:
    metadata:
      labels:
{{ $labels | toYaml | indent 8 }}
{{- if eq .Release.Namespace "istio-system"}}
        heritage: Tiller
        release: istio
        chart: gateways
{{- end }}
      annotations:
        sidecar.istio.io/inject: "false"
{{- if $gateway.podAnnotations }}
{{ toYaml $gateway.podAnnotations | indent 8 }}
{{ end }}
    spec:
{{- if .Values.global.priorityClassName }}
      priorityClassName: "{{ .Values.global.priorityClassName }}"
{{- end }}
{{- if .Values.global.proxy.enableCoreDump }}
      initContainers:
        - name: enable-core-dump
{{- if contains "/" .Values.global.proxy_init.image }}
          image: "{{ .Values.global.proxy_init.image }}"
{{- else }}
          image: "{{ .Values.global.hub }}/{{ .Values.global.proxy_init.image | default "proxy_init" }}:{{ .Values.global.tag }}"
```

**Comment by framled** [comment]
I think that I had the same issue using isitoctl 1.4.6

```yaml
apiVersion: install.istio.io/v1alpha2
kind: IstioControlPlane
spec:
  profile: sds
  gateways:
    components:
      egressGateway:
        enabled: true
```

I get the following error.

```bash
error   Error rendering the manifest: render error in "istio-egress/templates/serviceaccount.yaml": template: istio-egress/templates/serviceaccount.yaml:2:6: executing "istio-egress/templates/serviceaccount.yaml" at <($gateway.sds) and ($gateway.sds.enabled)>: can't give argument to non-function $gateway.sds
```

**Comment by howardjohn** [comment]
This is fixed in 1.5+ I believe
